Makefile

```diff
@@ -57,69 +18,8 @@ vet:
 lint: vet
 	cd $(ROOT_DIR) && golint
 
+docker:
+	./hack/docker.sh
 
-.PHONY: ./builds/$(GOOS)/$(GOARCH)/$(BIN_NAME)
-./builds/$(GOOS)/$(GOARCH)/$(BIN_NAME):
-	GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO_BUILD) -o ./builds/$(GOOS)/$(GOARCH)/$(BIN_NAME) .
-.PHONY: ./builds/$(GOOS)/$(GOARCH)/v$(GOARM)/$(BIN_NAME)
-./builds/$(GOOS)/$(GOARCH)/v$(GOARM)/$(BIN_NAME):
-	GOOS=$(GOOS) GOARCH=$(GOARCH) GOARM=$(GOARM) $(GO_BUILD) -o ./builds/$(GOOS)/$(GOARCH)/v$(GOARM)/$(BIN_NAME) .
-
-build-all-platforms: build-all-platforms-in-$(BUILD_IN)
-
-build-all-platforms-in-host:
-	$(MAKE) GOOS=linux   GOARCH=amd64       ./builds/linux/amd64/$(BIN_NAME)
-	$(MAKE) GOOS=linux   GOARCH=arm64       ./builds/linux/arm64/$(BIN_NAME)
-	$(MAKE) GOOS=linux   GOARCH=arm GOARM=7 ./builds/linux/arm/v7/$(BIN_NAME)
-	$(MAKE) GOOS=darwin  GOARCH=amd64       ./builds/darwin/amd64/$(BIN_NAME)
-	$(MAKE) GOOS=windows GOARCH=amd64       ./builds/windows/amd64/$(BIN_NAME)
-
-build-all-platforms-in-docker:
-	rm -rf ./builds
-	docker build --rm -t preflight-bin -f ./builder.dockerfile \
-		--build-arg oauth_client_id=$(OAUTH_CLIENT_ID) \
-		--build-arg oauth_client_secret=$(OAUTH_CLIENT_SECRET) \
-		--build-arg oauth_auth_server_domain=$(OAUTH_AUTH_SERVER_DOMAIN) \
-		.
-	docker create --rm --name=preflight-bin-container preflight-bin
-	docker cp preflight-bin-container:/go/github.com/jetstack/preflight/builds ./builds
-	docker rm preflight-bin-container
-	docker rmi preflight-bin
```

Comment:
We added this just to have more reproducibility in builds: i.e. if in my local machine I have docker 1.14 but I want to build with the same version that CI uses, I can use this and avoid installing another Go version on my system.

It is not a dealbreaker, but I think it is useful.

I am ok if this goes away if we gain some speed in CI.

Reply:
Lets see how it goes without it, won't be hard to add back in if we decide we need it again. Right now our builds aren't 100% reproducible anyway since we don't vendor our dependencies. To me it didn't seem necessary for the actual CI workflow, but if it's handy locally I can add it back